setup.py

```diff
@@ -37,6 +38,23 @@ def pypi_link(pkg_filename):
     return '/'.join(parts)
 
 
+_vendored_distutils_path = Path(here) / "setuptools" / "_distutils"
+
+
+def generate_distutils_stubs(destination: Path) -> None:
+    for path in _vendored_distutils_path.rglob("*.py"):
+        relative_path = path.relative_to(_vendored_distutils_path)
+        if relative_path.parts[0] == "tests":
+            continue
+        stub_path = (destination / relative_path).with_suffix(".pyi")
+        stub_path.parent.mkdir(exist_ok=True)
+        module = (
+            "setuptools._distutils."
+            + str(relative_path.with_suffix("")).replace(os.sep, ".")
+        ).removesuffix(".__init__")
```

Comment:
Alternative code for readability (but less "correct"):
```suggestion
        module = (
            ("setuptools._distutils." + str(relative_path.with_suffix("")))
            .replace(os.sep, ".")
            .removesuffix(".__init__")
        )
```